it/src/main/java/com/google/cloud/teleport/it/dataflow/DefaultDataflowClient.java

```diff
@@ -0,0 +1,399 @@
+    LOG.info("Received response:\n{}", formatForLogging(response));
+
+    Job job = response.getJob();
+    // Wait until the job is active to get more information
+    JobState state = waitUntilActive(project, region, job.getId());
+    job = getJob(project, region, job.getId());
+    return getJobInfo(options, state, job, /*runner*/ "Dataflow");
+  }
+
+  @Override
+  public JobInfo launchClassicTemplate(String project, String region, LaunchConfig options)
+      throws IOException {
+    checkState(
+        options.specPath() != null,
+        "Cannot launch a template job without specPath. Please specify specPath and try again!");
+    LOG.info("Getting ready to launch {} in {} under {}", options.jobName(), region, project);
+    LOG.info("Using the spec at {}", options.specPath());
+    LOG.info("Using parameters:\n{}", formatForLogging(options.parameters()));
+
+    LaunchTemplateParameters parameters =
+        new LaunchTemplateParameters()
+            .setJobName(options.jobName())
+            .setParameters(options.parameters());
+
+    LOG.info("Sending request:\n{}", formatForLogging(parameters));
+
+    LaunchTemplateResponse response =
+        client
+            .projects()
+            .locations()
+            .templates()
+            .launch(project, region, parameters)
+            .setGcsPath(options.specPath())
+            .execute();
+    LOG.info("Received response:\n{}", formatForLogging(response));
+
+    Job job = response.getJob();
+    // Wait until the job is active to get more information
+    JobState state = waitUntilActive(project, region, job.getId());
+    job = getJob(project, region, job.getId());
+    return getJobInfo(options, state, job, /*runner*/ "Dataflow");
+  }
+
+  @Override
+  public JobInfo launchJob(String project, String region, LaunchConfig options) throws IOException {
+    checkState(
+        options.sdk() != null,
+        "Cannot launch a dataflow job "
+            + "without sdk specified. Please specify sdk and try again!");
+    checkState(
+        options.executable() != null,
+        "Cannot launch a dataflow job "
+            + "without executable specified. Please specify executable and try again!");
+    LOG.info("Getting ready to launch {} in {} under {}", options.jobName(), region, project);
+    LOG.info("Using the executable at {}", options.executable());
+    LOG.info("Using parameters:\n{}", formatForLogging(options.parameters()));
+    // Create SDK specific command and execute to launch dataflow job
+    List<String> cmd = new ArrayList<>();
+    switch (options.sdk().toLowerCase()) {
+      case "java":
+        checkState(
+            options.classname() != null,
+            "Cannot launch a dataflow job "
+                + "without classname specified. Please specify classname and try again!");
+        cmd.add("java");
+        cmd.add("-cp");
+        cmd.add(options.executable());
+        cmd.add(options.classname());
```

Comment:
For Java calling the class directly will give more observability/stability on the tests, stracktrace, etc. Also just pointing out that the strategy here is different than Flex, as it doesn't require a staged template -- so it is validating the implementation, not the release/definitions as well.

On Python/Go - the plugin doesn't support them today, but that may be great to converge the strategies of IT + Performance in general.